crates/sparrow-runtime/src/execute/operation/scan.rs

```diff
@@ -233,16 +222,38 @@ impl ScanOperation {
                 .change_context(Error::internal_msg("failed to create stream reader"))?
                 .map_err(|e| e.change_context(Error::internal_msg("failed to read batch")))
                 .boxed();
+
                 input_stream
             }
         };
 
+        // Streams until the stop signal has been received
+        let input_stream = if let Some(mut stop_signal_rx) = stop_signal_rx {
+            input_stream
+                .take_until(async move {
```

Comment:
What are your thoughts between `take_until` vs. `take_while`? Here it may not matter as much since we are reading from the event stream, but the difference would be between inclusive or exclusive of the last event. 

Are we unable to use `take_while` because of the async move?

Reply:
yeah `take_while` didn't work because a) it requires a `fut` and b) we're using a `watch` channel, where the only future available is on `changed()`. So we need to wait _until_ the watch channel sees a new value